fenix/app/src/main/java/org/mozilla/fenix/shopping/state/ReviewQualityCheckAction.kt

```diff
@@ -9,4 +9,47 @@ import mozilla.components.lib.state.Action
 /**
  * Actions for review quality check feature.
  */
-sealed interface ReviewQualityCheckAction : Action
+sealed interface ReviewQualityCheckAction : Action {
+
+    /**
+     * Actions that are observed by middlewares.
+     */
+    sealed interface MiddlewareAction : ReviewQualityCheckAction
+
+    /**
+     * Actions that cause updates to state.
+     */
+    sealed interface UpdateAction : ReviewQualityCheckAction
+
+    /**
+     * Actions related to preferences.
+     */
+    sealed interface PreferencesMiddlewareAction : MiddlewareAction
+
+    /**
+     * Triggered when the store is initialized.
+     */
+    object Init : PreferencesMiddlewareAction
+
+    /**
+     * Triggered when the user has opted in or out of the review quality check feature.
+     */
+    data class OptInAction(
```

Comment:
Having `action` in the name is a bit redundant
```suggestion
    data class OptInChange(
```

Alternatively, we could have two actions
```suggestion
    object OptIn(
    object OptOut(
```